### PR TITLE
Cloudstack securityrule test refactor pr four

### PR DESCRIPTION
**Pull request by fgan1**
# Description
Refactoring the Cloudstack Security Rules Plugin context. 

# Proposed changes
- Refactoring "get Instance" tests context.

# Additional information
- This branch depends on a branch in the Fogbow Common (cloudstack-securityrules-test-refator)
Note: Maybe happen compiling error at PRs before PR5 in regards to Common Project because there were some changes in the PR5.

# PR Slices
[x] One: Refactoring in the Cloudstack Security Rules Plugin.
[x] Two: Migrating Cloudstack asynchronous operation to the Cloudstack Utils.
[x] Three: Refactoring "request Instance" tests context.
[x] Four: Refactoring "get Instance" tests context.
[] Five: Refactoring "delete instance" tests context.

# Reminding
- PRs order: 
develop < PR1 < PR2 < **PR3 < PR4** < PR5

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java [hunk 1]
```diff
@@ -97,6 +88,24 @@ public void deleteSecurityRule(String securityRuleId, @NotNull CloudStackUser cl
...
+    List<SecurityRuleInstance> doGetSecurityRules(@NotNull Order majorOrder,
+                                                  @NotNull CloudStackUser cloudStackUser)
+            throws FogbowException {
+
+        switch (majorOrder.getType()) {
+            case PUBLIC_IP:
+                String publicIpId = CloudStackPublicIpPlugin.getPublicIpId(majorOrder.getId());
+                return getFirewallRules(publicIpId, cloudStackUser);
+            case NETWORK:
+                return new ArrayList<>();
```
Comment (raonismaneoto) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: there isn't security rules for network in cloudstack?

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java [hunk 2]
```diff
@@ -378,6 +379,251 @@ public void testCheckRequestSecurityParametersWhenOrderTypeWrong()
...
+        String protocol = "unknown";
+
+        // exercise
+        SecurityRule.Protocol fogbowProtocol = this.plugin.getFogbowProtocol(protocol);
+
+        // verify
+        Assert.assertNull(fogbowProtocol);
+    }
+
     // # ------- old code --------- @
```
Comment (raonismaneoto) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java]: shouldn't it be removed?

Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java]: It was removed in the last PR(five).

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java (hunk 1 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: Yeap, there isn't it.

**Pull request merged**